pkg/reconciler/parallel/resources/subscription.go

```diff
@@ -57,19 +57,28 @@ func NewFilterSubscription(branchNumber int, p *v1.Parallel) *messagingv1.Subscr
 			},
 		},
 	}
-	if p.Spec.Branches[branchNumber].Filter != nil {
+	if p.Spec.Branches[branchNumber].Filter == nil {
```

Comment:
Should we just not create a subscription for a filter at all when there is no filter defined?

Reply:
Old behavior was like this:

- If `filter` is defined, use `subscription.subscriber` field as the filter AND use `subscriber.reply` as the parallel's ingress channel
- If `filter` is not defined, use `subscription.reply` as the parallel's ingress channel.

Before https://github.com/knative/eventing/issues/5756, if you leave `subscription.subscriber` empty and define something in `subscription.reply`, `subscription.reply` was used as the subscriber

With this PR, 

- If `filter` is defined, use `subscription.subscriber` field as the filter AND use `subscriber.reply` as the parallel's ingress channel --> same as before
- If `filter` is not defined, use `subscription.subscriber` as the parallel's ingress channel. --> changed